sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/ui/setup/SyncSetupFlowFragment.kt

```diff
@@ -0,0 +1,137 @@
+import com.duckduckgo.sync.impl.ui.setup.SyncSetupFlowViewModel.Command.FinishSetupFlow
+import com.duckduckgo.sync.impl.ui.setup.SyncSetupFlowViewModel.Command.RecoverSyncData
+import com.duckduckgo.sync.impl.ui.setup.SyncSetupFlowViewModel.Command.SyncAnotherDevice
+import com.duckduckgo.sync.impl.ui.setup.SyncSetupFlowViewModel.ViewMode
+import com.duckduckgo.sync.impl.ui.setup.SyncSetupFlowViewModel.ViewMode.InitialSetupScreen
+import com.duckduckgo.sync.impl.ui.setup.SyncSetupFlowViewModel.ViewMode.SyncAnotherDeviceScreen
+import com.duckduckgo.sync.impl.ui.setup.SyncSetupFlowViewModel.ViewState
+import javax.inject.*
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+
+@InjectWith(FragmentScope::class)
+class SyncSetupFlowFragment : DuckDuckGoFragment(R.layout.fragment_sync_setup) {
+    @Inject
+    lateinit var viewModelFactory: FragmentViewModelFactory
+
+    private val binding: FragmentSyncSetupBinding by viewBinding()
+
+    private var fragmentType: ViewMode = InitialSetupScreen
+
+    private val viewModel: SyncSetupFlowViewModel by lazy {
+        ViewModelProvider(this, viewModelFactory)[SyncSetupFlowViewModel::class.java]
+    }
+
+    private val listener: SetupFlowListener?
+        get() = activity as? SetupFlowListener
+
+    override fun onViewCreated(
+        view: View,
+        savedInstanceState: Bundle?,
+    ) {
+        super.onViewCreated(view, savedInstanceState)
+        observeUiEvents()
+        configureListeners()
+    }
+
+    private fun configureListeners() {
+        binding.closeIcon.setOnClickListener {
+            viewModel.onCloseClicked()
+        }
+    }
+
+    private fun observeUiEvents() {
+        viewModel
+            .viewState(fragmentType)
+            .flowWithLifecycle(lifecycle, Lifecycle.State.CREATED)
+            .onEach { viewState -> renderViewState(viewState) }
+            .launchIn(lifecycleScope)
+
+        viewModel
+            .commands()
+            .flowWithLifecycle(lifecycle, Lifecycle.State.STARTED)
+            .onEach { processCommand(it) }
+            .launchIn(lifecycleScope)
+    }
+
+    private fun renderViewState(viewState: ViewState) {
+        when (viewState.viewMode) {
+            InitialSetupScreen -> {
+                binding.contentIllustration.setImageResource(R.drawable.ic_sync_128)
+                binding.contentTitle.text = getString(R.string.sync_enable_sync_title)
+                binding.contentBody.text = getString(R.string.sync_enable_sync_content)
+                binding.footerPrimaryButton.text = getString(R.string.sync_enable_sync_primary_button)
+                binding.footerPrimaryButton.setOnClickListener {
+                    viewModel.onTurnOnSyncClicked()
+                }
+                binding.footerSecondaryButton.text = getString(R.string.sync_enable_sync_secondary_button)
+                binding.footerSecondaryButton.setOnClickListener {
```

Comment:
move this to `configureListeners()`

Reply:
I don't think we can move them into a `configureListeners()` since they change depending on the viewstate we are rendering, or I don't understand how.